### DISCUSSION TITLE
Include rewritten URLs into typedRoutes?

### DISCUSSION THREAD

**Discussion by tinleym** [vercel/next.js#58334] [App Router]
Does anyone know if it's possible to clue typedRoutes to rewritten URLs? Some, like those defined in middleware, seem like they'd be impossible to auto-generate.

**Comment by mutasim77** [comment]
Basically ```typedRoutes``` works by generating a hidden ```.d.ts``` file during ```next dev``` or ```next build```.  This means that ```.d.ts``` inside ```.next``` contains information only about the routes you wrote in your pages(for rewritten URLs, it has no idea).

**Reply by tinleym** [reply]
Right, I was wondering if there was a way to account for rewrites.
